sklearn/decomposition/_online_lda_fast.pyx

```diff
@@ -28,9 +30,11 @@ def mean_change(cnp.ndarray[ndim=1, dtype=floating] arr_1,
     return total / size
 
 
-def _dirichlet_expectation_1d(cnp.ndarray[ndim=1, dtype=floating] doc_topic,
-                              floating doc_topic_prior,
-                              cnp.ndarray[ndim=1, dtype=floating] out):
+def _dirichlet_expectation_1d(
+    floating[:] doc_topic,
```

Comment:
I think these should be const-qualified as well

Reply:
It seems we are modifying doc_topic and out in the function.
`for i in range(size):
        dt = doc_topic[i] + doc_topic_prior
        doc_topic[i] = dt
        total += dt`

Reply:
my bad, indeed, thanks